widgetsnbextension/src/manager.js

```diff
@@ -190,12 +188,21 @@ WidgetManager.set_state_callbacks = function (load_callback, save_callback, opti
 var url = [window.location.protocol, '//', window.location.host, window.location.pathname].join('');
 var key = 'widgets:' + url;
 WidgetManager.set_state_callbacks(function() {
-    if (localStorage[key]) {
-        return JSON.parse(localStorage[key]);
+    if (Jupyter.notebook.metadata.widgets) {
+        return return Promise.resolve(Jupyter.notebook.metadata.widgets.state);
     }
-    return {};
+    return Promise.resolve({});
 }, function(state) {
-    localStorage[key] = JSON.stringify(state);
+    Jupyter.notebook.metadata.widgets = {
+
+        // Persisted widget state version
+        version: '1.0.0',
```

Comment:
It could be the version of widgetsnbextension, because it's specific to the notebook. Really it's a separate version number though, like nbformat it may not change along with major releases. The purpose of the version number at the top level here is different than a version number per model. It's used to identify the storage format of the state.